src/annotator/notebook.js

```diff
@@ -1,147 +1,131 @@
-      this.frame?.remove();
-      this.frame = createNotebookFrame(this.options, this._groupId);
-      container.appendChild(this.frame);
-    }
+    render(
+      <NotebookOuter
+        eventBus={eventBus}
+        config={config}
+        disableScrolling={() => this.disableScrolling()}
+        enableScrolling={() => this.enableScrolling()}
+      />,
+      shadowRoot
+    );
   }
 
-  open() {
-    const container = this._initContainer();
-    this._update();
-    container.classList.add('is-open');
-    container.style.display = '';
-
-    // The overflow CSS property is set to hidden to prevent scrolling of the guest page,
-    // while the notebook is shown as modal. It is restored on the close method.
-    // I believe this hack only works if this.element points to document.body of the guest page.
-    this.originalOverflowStyle = this.element.style.overflow;
+  /**
+   * The overflow CSS property is set to hidden to prevent scrolling of the host page,
+   * while the notebook is shown as modal. It is restored on the close method.
+   * This hack only works if this.element points to document.body of the guest page.
+   */
+  disableScrolling() {
     this.element.style.overflow = 'hidden';
   }
 
-  close() {
-    if (this.container) {
-      this.container.classList.remove('is-open');
-      this.container.style.display = 'none';
-    }
-    this.element.style.overflow = /** @type {string} */ (this
-      .originalOverflowStyle);
+  enableScrolling() {
+    this.element.style.overflow = this._originalOverflowStyle;
   }
 
   destroy() {
     this._outerContainer.remove();
-    this._emitter.destroy();
   }
+}
 
-  _initContainer() {
-    if (this.container) {
-      return this.container;
-    }
+/**
+ * Create the iframe that will load the notebook application.
+ * @param {object} props
+ * @param {Record<string, any>} props.config
+ * @param {string|null} props.groupId
+ */
+function NotebookIframe({ config, groupId }) {
+  const notebookConfig = createSidebarConfig(config);
+  // Explicity set the "focused" group
+  notebookConfig.group = groupId;
+  const configParam = encodeURIComponent(JSON.stringify(notebookConfig));
+  const notebookAppSrc = `${config.notebookAppUrl}#config=${configParam}`;
+
+  return (
+    <iframe
+      title={'Hypothesis annotation notebook'}
+      className="Notebook__iframe"
+      // Enable media in annotations to be shown fullscreen
+      allowFullScreen
+      src={notebookAppSrc}
+    />
+  );
+}
 
-    const shadowRoot = createShadowRoot(this._outerContainer);
-    this.container = document.createElement('div');
-    this.container.style.display = 'none';
-    this.container.className = 'notebook-outer';
-    shadowRoot.appendChild(this.container);
+/**
+ * @param {object} props
+ * @param {import('../types/annotator').EventBus} props.eventBus
+ * @param {Record<string, any>} props.config
+ * @param {VoidFunction} props.disableScrolling - disables scrolling of document.body
```

Comment:
When looking just at these prop docs, it's unclear what the functions `disableScrolling` and `enableScrolling` are meant to be responsible for. When I look at the implementation of the functions that get passed, it's perfectly clear.

I'm wondering if this might be slightly over-abstracted here. Not sure. In any case, a bit of increased clarity would be helpful.

Reply:
I have removed disabled/enableScrolling